L1Trigger/L1TTrackerDTC/src/DTC.cc

```diff
@@ -0,0 +1,145 @@
+#include "L1Trigger/L1TTrackerDTC/interface/DTC.h"
+#include "L1Trigger/L1TTrackerDTC/interface/Settings.h"
+#include "L1Trigger/L1TTrackerDTC/interface/Module.h"
+#include "L1Trigger/L1TTrackerDTC/interface/Stub.h"
+
+using namespace std;
+using namespace edm;
+
+namespace L1TTrackerDTC {
+
+  DTC::DTC(Settings* settings, const int& dtcId, const std::vector<Module*>& modules, const int& nStubs)
+      : settings_(settings),                            // helper class to store configurations
+        region_(dtcId / settings->numDTCsPerRegion()),  // outer tracker detector region [0-8]
+        board_(dtcId % settings->numDTCsPerRegion()),   // outer tracker dtc id in region [0-23]
+        modules_(modules)                               // container of sensor modules connected to this DTC
+  {
+    stubs_.reserve(nStubs);  // container of dynamic allocated stubs on this DTC
+  }
+
+  DTC::~DTC() {
+    for (auto& stub : stubs_)
+      delete stub;
```

Comment:
do all the stubs really need to be allocated dynamically? this implies some performance penalty

Reply:
static allocation now